homeassistant/components/onewire/config_flow.py

```diff
@@ -0,0 +1,196 @@
+    {
+        vol.Required(CONF_MOUNT_DIR, default=DEFAULT_SYSBUS_MOUNT_DIR): str,
+    }
+)
+
+
+async def validate_input_owserver(hass: HomeAssistantType, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA_OWSERVER with values provided by the user.
+    """
+
+    hub = OneWireHub(hass)
+
+    host = data[CONF_HOST]
+    port = data[CONF_PORT]
+    if not await hub.can_connect(host, port):
+        raise CannotConnect
+
+    # Return info that you want to store in the config entry.
+    return {"title": host}
+
+
+def is_duplicate_owserver_entry(hass: HomeAssistantType, user_input):
+    """Check existing entries for matching host and port."""
+    for config_entry in hass.config_entries.async_entries(DOMAIN):
+        if (
+            config_entry.data[CONF_TYPE] == CONF_TYPE_OWSERVER
+            and config_entry.data[CONF_HOST] == user_input[CONF_HOST]
+            and config_entry.data[CONF_PORT] == str(user_input[CONF_PORT])
+        ):
+            return True
+    return False
+
+
+async def validate_input_mount_dir(hass: HomeAssistantType, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA_MOUNTDIR with values provided by the user.
+    """
+    hub = OneWireHub(hass)
+
+    mount_dir = data[CONF_MOUNT_DIR]
+    if not await hub.is_valid_mount_dir(mount_dir):
+        raise InvalidPath
+
+    # Return info that you want to store in the config entry.
+    return {"title": mount_dir}
+
+
+class OneWireFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle 1-Wire config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize 1-Wire config flow."""
+        self.onewire_config = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle 1-Wire config flow start.
+
+        Let user manually input configuration.
+        """
+        errors = {}
+        if user_input is not None:
+            self.onewire_config.update(user_input)
+            if CONF_TYPE_OWSERVER == user_input[CONF_TYPE]:
+                return await self.async_step_owserver()
+            if CONF_TYPE_SYSBUS == user_input[CONF_TYPE]:
+                return await self.async_step_mount_dir()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=DATA_SCHEMA_USER,
+            errors=errors,
+        )
+
+    async def async_step_owserver(self, user_input=None):
+        """Handle OWServer configuration."""
+        errors = {}
+        if user_input:
+            # Prevent duplicate entries
+            if is_duplicate_owserver_entry(self.hass, user_input):
+                return self.async_abort(reason="already_configured")
+
+            self.onewire_config.update(user_input)
+
+            try:
+                info = await validate_input_owserver(self.hass, user_input)
+
+                return self.async_create_entry(
```

Comment:
Please move this to an `else:` block. We only want the line that can raise in the try... except block.

Reply:
Updated in d29384e